app/components/widgets/AlertBox.vue

```diff
@@ -0,0 +1,298 @@
+      <v-form-group :title="$t('Text Color')" type="color" v-model="selectedVariation.settings.message.color" />
+      <v-form-group :title="$t('Allow Twitch Emojis?')" type="bool" v-model="selectedVariation.settings.message.allowEmojis" />
+    </div>
+    <div v-if="selectedVariation.settings.tts">
+      <span>{{ $t('Text to Speech') }}</span>
+      <v-form-group :title="$t('Enable TTS?')" type="bool" v-model="selectedVariation.settings.tts.enabled" />
+      <v-form-group :title="$t('Min. Amount to Read')" type="number" v-model="selectedVariation.settings.tts.minAmount" />
+      <v-form-group :title="$t('Volume')" type="slider" v-model="selectedVariation.settings.tts.volume" :metadata="{ min: 0, max: 100 }" />
+      <v-form-group :title="$t('Language')" type="list" v-model="selectedVariation.settings.tts.language" />
+      <v-form-group :title="$t('Spam Security')" type="slider" v-model="selectedVariation.settings.tts.security" />
+    </div>
+  </validated-form>
+  <validated-form slot="animation-properties" @input="save()" v-if="selectedVariation">
+    <v-form-group :title="$t('Show Animation')" type="animation" v-model="selectedVariation.settings.showAnimation" :metadata="{ filter: 'in' }" />
+    <v-form-group :title="$t('Hide Animation')" type="animation" v-model="selectedVariation.settings.hideAnimation" :metadata="{ filter: 'out' }" />
+    <v-form-group :title="$t('Alert Duration')" type="slider" v-model="selectedVariation.settings.duration" :metadata="{ min: 2, max: 300 }" />
+    <v-form-group :title="$t('Text Animation')" type="animation" v-model="selectedVariation.settings.text.animation"  :metadata="{ filter: 'text' }" v-if="selectedVariation.settings.text" />
+    <v-form-group :title="$t('Text Delay')" type="slider" v-model="selectedVariation.settings.textDelay" :metadata="{ min: 0, max: 60 }" />
+  </validated-form>
+  <validated-form slot="alert-properties" @input="save()" v-if="selectedVariation">
+    <v-form-group :title="$t('Min. Amount to Trigger Alert')" type="number" v-model="minTriggerAmount" v-if="['donations', 'bits', 'hosts', 'raids'].includes(selectedAlert)" />
+    <v-form-group :title="$t('Min. Amount to Show in Recent Events')" type="number" v-model="minRecentEvents" v-if="['donations', 'hosts'].includes(selectedAlert)" />
+    <div v-if="selectedId !== 'default'">
+      <v-form-group :title="$t('Variation Condition')" type="list" v-model="selectedVariation.condition" :metadata="{ options: conditions }" />
+      <v-form-group :title="$t('Variation Frequency')" v-if="selectedVariation.condition === 'RANDOM'">
+        <frequency-input v-model="selectedVariation.conditionData" />
+      </v-form-group>
+      <v-form-group :title="$t('Variation Amount')" type="number" v-model="selectedVariation.conditionData" v-if="selectedVariation.condition !== 'RANDOM'" />
+    </div>
+  </validated-form>
+</widget-editor>
+</template>
```

Comment:
Extracting metadata to the service probably the best way how we should implement forms.
Motivation 1:
As a third-party developer, I can create my own version of alert box settings. So I would prefer to fetch all metadata for `alert box` fields via an API request to the widget service.
Motivation 2:
Metadata can contain constraints like min and max values for number inputs. I can use it in tests to check that validations work. There are no ways to fetch this data from Vue components now in our tests

So can you please extract the metadata into service like here https://github.com/stream-labs/streamlabs-obs/pull/968/files#diff-161edac13f57f342c71dfd4a733621d0R52

I'm doing this for each widget that I'm writing tests for

Reply:
looks like the `formMetadata` helper was added in that branch so i won't be able to do that until that branch gets merged in

Reply:
`formMetadata` just adds a `name` property for each input. You can use a simple object instead for now. I can wrap it later